src/sentry/replaystore/base.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import abc
+from dataclasses import dataclass
+from datetime import datetime
+from enum import IntEnum
+from threading import local
+from typing import Any, Dict, List, Tuple, cast
+
+from sentry.utils import json
+from sentry.utils.services import Service
+from sentry.utils.strings import decompress
+
+
+class ReplayNotFound(ValueError):
+    pass
+
+
+class ReplayDataType(IntEnum):
+    INIT = 1
+    EVENT = 2
+    PAYLOAD = 3
+
+    @staticmethod
+    def should_be_encoded() -> list[ReplayDataType]:
+        return [ReplayDataType.INIT, ReplayDataType.EVENT]
+
+
+@dataclass(frozen=True)
+class ReplayContainer:
+    id: str
+    init: Dict[Any, Any]
+    events: List[Dict[Any, Any]]
+    payloads: List[Dict[Any, Any]]
+
+
+json_dumps = json.JSONEncoder(
+    separators=(",", ":"),
+    sort_keys=True,
+    skipkeys=False,
+    ensure_ascii=True,
+    check_circular=True,
+    allow_nan=True,
+    indent=None,
+    encoding="utf-8",
+    default=None,
+).encode
+
+json_loads = json._default_decoder.decode
+
+
+class ReplayStore(abc.ABC, local, Service):
+    KEY_DELIMETER = ":"
+
+    __all__ = (
+        "set",
+        "get_replay",
+    )
+
+    def get_replay(self, replay_id: str) -> ReplayContainer | None:
+
+        key = f"{replay_id}{self.KEY_DELIMETER}"
+
+        try:
+            init, events, payloads = self._get_all_events_for_replay(key)
+        except ReplayNotFound:
+            return None
+        replay = ReplayContainer(replay_id, init, events, payloads)
+        return replay
+
+    def set(
+        self, replay_init_id: str, data: Any, replay_data_type: ReplayDataType, timestamp: datetime
```

Comment:
Can we more strictly type `data`?

If this is JSON, we could at least use `JSONData`. But if we have an actual schema here it might be nice to use a typeddict

Reply:
i've refactored to use two separate functions since payloads are bytes and the replay_events are more structured, of which we can add further types down the line once the schema is ironed out more.